### PR TITLE
[network-key] change name from `master` to `network`

### PR DESCRIPTION
**Pull request by jwhui**
To conform to [SPEC-982](https://threadgroup.atlassian.net/jira/core/projects/SPEC/issues/SPEC-982).

Depends-On: openthread/ot-br-posix#889
Depends-On: openthread/pyspinel#116
Depends-On: openthread/ot-ns#132

### CODE REVIEW

File: tools/harness-thci/OpenThread_WpanCtl.py [hunk 1]
```diff
@@ -2347,8 +2347,8 @@ def MGMT_ACTIVE_SET(
 
             if xMasterKey is not None:
```
Comment (simonlingoogle) [inline tools/harness-thci/OpenThread_WpanCtl.py:2348]: Should we fix `xMasterKey` too?

**Comment by simonlingoogle** [comment]
Found `master` in these files:
- examples/platforms/simulation/README.md
- script/check-api-version
- tests/scripts/thread-cert/
- tools/harness-thci

@jwhui

File: tools/harness-thci/OpenThread_WpanCtl.py (hunk 1 above)
Comment (jwhui) [reply tools/harness-thci/OpenThread_WpanCtl.py:2348]: Is `MGMT_ACTIVE_SET()` called by test harness? Will updating `xMasterKey` affect compatibility with test harness? If not, then we should update this as well.

**Comment by jwhui** [comment]
> Found `master` in these files:
> 
> * examples/platforms/simulation/README.md

Fixed.

> * script/check-api-version

Fixed.

> * tests/scripts/thread-cert/

Changing `thread_meshcop.tlv.master_key` breaks compatibility with dissector.

> * tools/harness-thci

Will changing `xMasterKey` break compatibility with test harness?

File: src/core/meshcop/dataset.hpp [hunk 2]
```diff
@@ -857,13 +857,13 @@ class Dataset
      * This method applies the Active or Pending Dataset to the Thread interface.
      *
      * @param[in]  aInstance           A reference to the OpenThread instance.
-     * @param[out] aIsMasterKeyUpdated A pointer to where to place whether master key was updated.
+     * @param[out] aIsNetworkKeyUpdated A pointer to where to place whether network key was updated.
```
Comment (abtink) [inline src/core/meshcop/dataset.hpp:860]: nit: align with `aInstance`

**Review by abtink — APPROVED** [review]
Looks great! Thanks for doing this.
Just found one small alignment in comment.

**Review by bukepo — APPROVED** [review]

**Review by librasungirl — APPROVED** [review]

**Comment by simonlingoogle** [comment]
> > * tests/scripts/thread-cert/
> 
> Changing `thread_meshcop.tlv.master_key` breaks compatibility with dissector.

Yes. We can't fix it for now.


> > * tools/harness-thci
> 
> Will changing `xMasterKey` break compatibility with test harness?

It will. We can't fix it for now.

**Review by simonlingoogle — APPROVED** [review]
LGTM :+1:

**Pull request merged**